sdk/services/exchange.ts

```diff
@@ -734,46 +723,42 @@ export default class ExchangeService {
 	}
 
 	public async getRedeemableDeprecatedSynths() {
-		if (!this.sdk.context.contracts?.SynthRedeemer) {
+		const { SynthRedeemer } = this.sdk.context.contracts;
+		const { SynthRedeemer: Redeemer } = this.sdk.context.multicallContracts;
+
+		if (!SynthRedeemer || !Redeemer) {
 			throw new Error(sdkErrors.UNSUPPORTED_NETWORK);
 		}
 
-		const SynthRedeemer = this.sdk.context.contracts.SynthRedeemer.connect(
-			this.sdk.context.provider
-		);
+		const { walletAddress } = this.sdk.context;
 
 		const synthDeprecatedFilter = SynthRedeemer.filters.SynthDeprecated();
 		const deprecatedSynthsEvents = await SynthRedeemer.queryFilter(synthDeprecatedFilter);
 		const deprecatedProxySynthsAddresses: string[] =
 			deprecatedSynthsEvents.map((e) => e.args?.synth).filter(Boolean) ?? [];
 
-		const Redeemer = new EthCallContract(SynthRedeemer.address, SynthRedeemerABI);
-
 		const symbolCalls = [];
 		const balanceCalls = [];
 
 		for (const addr of deprecatedProxySynthsAddresses) {
 			symbolCalls.push(getProxySynthSymbol(addr));
-			balanceCalls.push(Redeemer.balanceOf(addr, this.sdk.context.walletAddress));
+			balanceCalls.push(Redeemer.balanceOf(addr, walletAddress));
 		}
 
 		const [deprecatedSynths, balanceData] = (await Promise.all([
 			this.sdk.context.multicallProvider.all(symbolCalls),
```

Comment:
We could combine and then split these to have them in one call

Reply:
Done